packages/vm/lib/evm/opcodes/functions.ts

```diff
@@ -469,15 +469,18 @@ export const handlers: Map<number, OpHandler> = new Map([
       const [memOffset, dataOffset, dataLength] = runState.stack.popN(3)
 
       subMemUsage(runState, memOffset, dataLength)
-      runState.eei.useGas(
-        new BN(runState._common.param('gasPrices', 'copy')).imul(divCeil(dataLength, new BN(32)))
-      )
 
-      const data = getDataSlice(runState.eei.getCallData(), dataOffset, dataLength)
-      const memOffsetNum = memOffset.toNumber()
-      const dataLengthNum = dataLength.toNumber()
-      runState.memory.extend(memOffsetNum, dataLengthNum)
-      runState.memory.write(memOffsetNum, dataLengthNum, data)
+      if (!dataLength.eqn(0)) {
```

Comment:
The `subMemUsage()` call here is out of the condition, can this remain consistent (and correct 😜 )?

Reply:
Will make these 4 functions consistent. FYI, in case data length is 0, then `subMemGas` is also no-op:

```typescript
export function subMemUsage(runState: RunState, offset: BN, length: BN) {
  // YP (225): access with zero length will not extend the memory
  if (length.isZero()) return
  // (...)
```